src/main/scala/tilelink/RegisterRouter.scala

```diff
@@ -42,6 +43,13 @@ case class TLRegisterNode(
       s"TLRegisterNode addresses (${address}) must be aligned to its size ${size}")
   }
 
+  def getMemory(name: String, omRegMap : Option[OMRegisterMap], resourceBindingsMap: ResourceBindingsMap): Seq[OMMemoryRegion] = {
+    val resourceBindings = resourceBindingsMap.map.get(device)
+    assert(device != None, s"Device is null : ${device.label}")
```

Comment:
What's going on with this assertion? `device` is a `Device`, not an `Option[Device]`, so this check will never work. Note that `null` and `None` are two different concepts in Scala. `None` is one of the two implementations of the `Option` abstract class. `null` is the same thing as the Java `null`, and it's usually only encountered when interoperating with Java code.